app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/MongoAbstractCustomizer.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.connector.mongo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.mongodb.client.result.DeleteResult;
+import com.mongodb.client.result.UpdateResult;
+import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.stream.Collectors.toList;
+
+public abstract class MongoAbstractCustomizer implements ComponentProxyCustomizer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoAbstractCustomizer.class);
+
+    /**
+     * Used to convert any result MongoOperation (either {@link DeleteResult} or {@link UpdateResult}
+     * to a {@link Long}
+     * @param exchange
+     */
+    protected void convertMongoResultToLong(Exchange exchange) {
```

Comment:
Methods like these can be made static, that also makes them a bit easier to test and it clearly shows that they're side effect free. Also ErrorProne's new severity fails the build if a method can be made static. Try the `autofix` Maven profile I think it can refactor these for you, you might need to run it a couple of times as it usually fails on the first file.